imap_processing/cdf/cdf_attribute_manager.py

```diff
@@ -236,10 +238,22 @@ def get_variable_attributes(self, variable_name: str) -> dict:
         Returns
         -------
         dict
-            I have no idea todo check.
+            Information containing specific variable attributes
+            associated with "variable_name".
         """
-        # TODO: Create a variable attribute schema file, validate here
-        if variable_name in self.variable_attributes:
-            return self.variable_attributes[variable_name]
-        # TODO: throw an error?
-        return {}
+        output = dict()
+        for attr_name in self.variable_attribute_schema.keys():
+            if variable_name in self._variable_attributes:
+                output = self._variable_attributes[variable_name]
+            elif (
+                variable_name is not None
```

Comment:
I don't think `variable_name` should ever be None, so this check is not needed. Actually, this whole `elif` block should be removed, because the person calling get_variable_attributes should always provide variable_name. 

This was necessary in get_global_attributes because the instrument_id is an optional parameter, so it can be none.

Reply:
It looks like this is why your tests are failing too!